src/Ipns.ts

```diff
@@ -0,0 +1,49 @@
+import fetchPonyfill from 'fetch-ponyfill'
+
+import { CheckBase } from './CheckBase'
+import { IPNS_PATH_TO_TEST } from './constants'
+import type { GatewayNode } from './GatewayNode'
+
+import { Log } from './Log'
+
+const { fetch } = fetchPonyfill()
+
+const log = new Log('Ipns')
+
+class IPNSCheck extends CheckBase implements Checkable {
+  _className = 'Ipns'
+  _tagName = 'div'
+  constructor (protected parent: GatewayNode) {
+    super(parent, 'div', 'Ipns')
+  }
+
+  async check (): Promise<void> {
+    const now = Date.now()
+    const gatewayUrl = new URL(this.parent.gateway)
+    gatewayUrl.pathname = IPNS_PATH_TO_TEST
+    const testUrl = `${gatewayUrl.href}?now=${now}`
```

Comment:
Since gateway URLs are hard coded with `/ipfs/`, we need to parse URLs and override the path to `/ipns/`.